File: shoring/fxapp/src/main/java/com/panopset/fxapp/FieldHelper.java
package com.panopset.fxapp;

import javafx.scene.control.PasswordField;
import javafx.scene.control.TextInputControl;

public enum FieldHelper {
  
  INSTANCE;
  
  public static boolean isPasswordField(TextInputControl field) {
    return INSTANCE.isPwd(field);
  }
  
  private boolean isIdAvailable(TextInputControl field) {
    return field.getId() != null;
  }

  private boolean isPwd(TextInputControl field) {
    if (isIdAvailable(field)) {
      return field instanceof PasswordField || field.getId().toLowerCase().contains("pwdshow");
    }
    return false;
  }

}

File: code/shoring/fxapp/src/main/kotlin/com/panopset/fxapp/FieldHelper.kt
package com.panopset.fxapp

import javafx.scene.control.PasswordField
import javafx.scene.control.TextInputControl
import java.util.*

private fun isIdAvailable(field: TextInputControl): Boolean {
    return field.id != null
}

private fun isPwd(field: TextInputControl): Boolean {
    return if (isIdAvailable(field)) {
        field is PasswordField || field.id.lowercase(Locale.getDefault()).contains("pwdshow")
    } else false
}

fun isPasswordField(field: TextInputControl): Boolean {
    return isPwd(field)
}
